apps/ca.c

```diff
@@ -1412,44 +1411,45 @@ static int do_body(X509 **xret, EVP_PKEY *pkey, X509 *x509,
 
     name = X509_REQ_get_subject_name(req);
     for (i = 0; i < X509_NAME_entry_count(name); i++) {
-        ne = X509_NAME_get_entry(name, i);
-        str = X509_NAME_ENTRY_get_data(ne);
-        obj = X509_NAME_ENTRY_get_object(ne);
+        ne = X509_NAME_get0_entry(name, i);
+        str = X509_NAME_ENTRY_get0_data(ne);
+        obj = X509_NAME_ENTRY_get0_object(ne);
 
         if (msie_hack) {
+            /* Dirty ugly cast to make the msie_hack compile!! */
```

Comment:
I don't like this at all. We need to retain a getter for a  mutable `ASN1_STRING` until we can sort this out.